api/app/src/providers/oauth2.ts

```diff
@@ -183,10 +183,19 @@ export class OAuth2DefaultImpl implements OAuth2 {
     params?: { [k: string]: string | number }
   ): Promise<T> {
     try {
-      const resp = await axios.get(endpoint, {
-        headers: {
+      let headers;
+      if (params?.locale) {
+        headers = {
+          Authorization: `Bearer ${access_token}`,
+          "Accept-Language": params.locale,
+        };
+      } else {
+        headers = {
           Authorization: `Bearer ${access_token}`,
-        },
+        };
+      }
```

Comment:
assuming you have `extraHeaders`, you can do this like:

```
headers = {
          Authorization: `Bearer ${access_token}`,
          ...(extraHeaders && extraHeaders),
        };
```

Note the `...` operator will spread the extraHeaders into the headers -> play around with this to get the feel of it